core/trino-main/src/test/java/io/trino/block/BenchmarkRowBlockBuilder.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.block;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.block.RowBlockBuilder;
+import io.trino.spi.block.SingleRowBlockWriter;
+import io.trino.spi.type.RowType;
+import io.trino.spi.type.Type;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.CompilerControl;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.profile.GCProfiler;
+
+import java.util.List;
+
+import static io.trino.jmh.Benchmarks.benchmark;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+
+@SuppressWarnings("MethodMayBeStatic")
+@State(Scope.Thread)
+@OutputTimeUnit(MICROSECONDS)
+@Fork(1)
+@Warmup(iterations = 15)
+@Measurement(iterations = 15)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkRowBlockBuilder
+{
+    @Benchmark
+    @CompilerControl(CompilerControl.Mode.DONT_INLINE)
```

Comment:
why? I don't see this in other benchmarks

Reply:
Becasue we want to be ensure that tested alloction would not esacped (by Escape Analysis)

Reply:
> Becasue we want to be ensure that tested alloction would not esacped (by Escape Analysis)

Why don't we want it to be escaped? In practice it might happen

Reply:
Yes, but possibly it is easier for JVM to make some optimization in "isolated" and "small" environment in comparison to real environment. The first goal of this pull request was to elide the allocation. This is why I wanted to be ensured that it is no escaped. But, if you want I can remove it.

Reply:
We don't use `CompilerControl.Mode.DONT_INLINE` other benchmarks, so I wanted to understand why we would need it.

> Yes, but possibly it is easier for JVM to make some optimization in "isolated" and "small" environment in comparison to real environment.

How `DONT_INLINE` affects that? Even in large production environments, JIT will still optimize relatively small % of code (tight loops)

> The first goal of this pull request was to elide the allocation

It's mean to improve performance. Reduced allocation rate is just another benefit.

Reply:
> How DONT_INLINE affects that? Even in large production environments, JIT will still optimize relatively small % of code (tight loops)
AFAIK I know if we block the inlining of the method in our case then JVM should not be able to make escape analysis.